tests/functional_tests/ffront_tests/test_execution.py

```diff
@@ -810,3 +810,121 @@ def solve_tridiag(
+    num_edges = reduction_setup.num_edges
+
+    a = np_as_located_field(Edge)(2 * np.ones((num_edges,)))
+    b = np_as_located_field(Edge)(2 * np.ones((num_edges,)))
+    out = np_as_located_field(Edge)(np.zeros((num_edges,)))
+
+    left = 2.0
+    right = 3.0
+
+    @field_operator
+    def ternary_field_op(
+        a: Field[[Edge], float], b: Field[[Edge], float], left: float, right: float
+    ) -> Field[[Edge], float]:
+        c = a + b if left < right else b
+        return c
+
+    ternary_field_op(a, b, left, right, out=out, offset_provider={})
+    e = np.asarray(a) + np.asarray(b) if left < right else np.asarray(b)
+    np.allclose(e, out)
+
+    @field_operator
+    def ternary_field_op_scalars(left: float, right: float) -> Field[[Edge], float]:
+        return broadcast(3.0, (Edge,)) if left < right else broadcast(4.0, (Edge,))
+
+    ternary_field_op_scalars(left, right, out=out, offset_provider={})
+    e = np.full(e.shape, 3.0) if left < right else e
+    np.allclose(e, out)
+
+
+def test_ternary_operator_tuple(reduction_setup):
+    Edge = reduction_setup.Edge
+    num_edges = reduction_setup.num_edges
+    a = np_as_located_field(Edge)(np.ones((num_edges,)))
+    b = np_as_located_field(Edge)(2 * np.ones((num_edges,)))
+    out_1 = np_as_located_field(Edge)(np.zeros((num_edges,)))
+    out_2 = np_as_located_field(Edge)(np.zeros((num_edges,)))
+
+    left = 2.0
+    right = 3.0
+
+    @field_operator
+    def ternary_field_op(
+        a: Field[[Edge], float], b: Field[[Edge], float], left: float, right: float
+    ) -> tuple[Field[[Edge], float], Field[[Edge], float]]:
+        c, d = (a, b) if left < right else (b, a)
+        return c, d
+
+    # TODO(tehrengruber): directly call field operator when the generated programs support `out` being a tuple
+    @program
+    def ternary_field(
+        a: Field[[Edge], float],
+        b: Field[[Edge], float],
+        left: float,
+        right: float,
+        out_1: Field[[Edge], float],
+        out_2: Field[[Edge], float],
+    ):
+        ternary_field_op(a, b, left, right, out=(out_1, out_2))
+
+    ternary_field(a, b, left, right, out_1, out_2, offset_provider={})
+
+    e, f = (np.asarray(a), np.asarray(b)) if 2 < 3 else (np.asarray(b), np.asarray(a))
+    np.allclose(e, out_1)
+    np.allclose(f, out_2)
+
+
+def test_ternary_builtin_neighbor_sum(reduction_setup):
+    rs = reduction_setup
+    Edge = rs.Edge
+    Vertex = rs.Vertex
+    V2EDim = rs.V2EDim
+    V2E = rs.V2E
+
+    num_vertices = rs.num_vertices
+    num_edges = rs.num_edges
+
+    a = np_as_located_field(Edge)(np.ones((num_edges,)))
+    b = np_as_located_field(Edge)(2 * np.ones((num_edges,)))
+    out = np_as_located_field(Vertex)(np.zeros((num_vertices,)))
+
+    @field_operator
+    def ternary_reduce(a: Field[[Edge], float], b: Field[[Edge], float]) -> Field[[Vertex], float]:
+        # line below does not work
+        # f = neighbor_sum(b(V2E) if 2 < 3 else a(V2E), axis=V2EDim)
```

Comment:
Maybe this fails because of the `IfExpr` typo in the lowering. In any case we should fix this before this PR is merged. As a general note: If a test fails you can use `pytest.xskip` together with a comment on the reason for the failure and potential solutions such that when someone revisits the test he can pick up where you left.

Reply:
I added some lines in the type_deduction for this part. The issue there was with the local dimensions